src/components/search/UseSearch.jsx

```diff
@@ -31,16 +33,26 @@ const useSearch = () => {
     } catch (error) {
       console.error('Failed to fetch classes:', error.message);
     } finally {
-      setLoading(false);
+      setTimeout(() => {
+        setLoading(false);
+      }, 200);
     }
   };
+  //this is the way it works for both MainPage(DesktopNav) and SearchPage page
+
+  useEffect(() => {
```

Comment:
Both these functions are doing the same things except one relies on the value of `search`. This is set up to run regardless of the value of `search`.

Reply:
I totally agree with you! It should work regardless of search  value and for the search page it works! BUT if I remove 

` useEffect(() => {
  if (search) {
     fetchClasses(search, currentPage);
    }
   }, [search, currentPage, category]);` 
it does not  work for main page. And I don't know how to fix it

Reply:
What doesn't work? The search functionality?